tests/main.test.ts

```diff
@@ -8,13 +8,38 @@ beforeEach(async () => {
   process.env.COREPACK_HOME = npath.fromPortablePath(await xfs.mktempPromise());
 });
 
+it(`should refuse to download a package manager if the hash doesn't match`, async () => {
+  await xfs.mktempPromise(async cwd => {
+    await xfs.writeJsonPromise(ppath.join(cwd, `package.json` as Filename), {
+      packageManager: `yarn@1.22.4+sha1.deadbeef`,
+    });
+
+    await expect(runCli(cwd, [`yarn`, `--version`])).resolves.toMatchObject({
+      exitCode: 1,
+      stdout: /Mismatch hashes/,
+    });
+  });
+});
+
 const testedPackageManagers: Array<[string, string]> = [
   [`yarn`, `1.22.4`],
+  [`yarn`, `1.22.4+md5.faf483d50aa8ccbdc802efa0cac5d4d3`],
```

Comment:
nit: we don't test `sha1`, which is what #137 uses

Reply:
Makes sense, I've switched to SHA1 and SHA224 in the tests as those are more credible options for hash algorithms.